deploy/kubernetes/elastic-agent-kustomize/default/README.md

```diff
@@ -0,0 +1,71 @@
+# Kustomize Templates
+
+The list below includes the official [kustomize](https://github.com/kubernetes-sigs/kustomize) templates to run them in Kubernetes:
+
+Agent Scenario | Description
+---- | ----
+[Elastic Agent managed - Default ](./elastic-agent-managed/) | Default Elastic Agent managed by Fleet setup. Kube-state-metrics (KSM) is installed automatically.
+[Elastic Agent standalone Default ](./elastic-agent-standalone/) | Default Standalone Elastic Agent setup. Kube-state-metrics (KSM) is installed automatically.
+
+## Using above templates
+
+Users can clone this repository to use the provided kustomize templates.
+
+For *Managed Elastic Agent*, please update the following variables inside main kustomization.yaml:
+
+- %FLEET_URL%: Fleet Server URL to enroll the Elastic Agent into. FLEET_URL can be found in Kibana, go to Management > Fleet > Settings
+- %ENROLLMENT_TOKEN%: Elasticsearch API key used to [enroll Elastic Agents](https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html#fleet-enrollment-tokens) in Fleet. *This should be encoded as base64 value because it will be stored as Kubernetes secret*
+
+Eg.
+
+```yaml
+secretGenerator:
+  - name: elastic-agent-creds
+    literals:
+      - enrollment_token=%ENROLLMENT_TOKEN%
+```
+
+Comm
+
+For *Standalone Elastic Agent*, please update the following secrets inside main [kustomization.yaml](./elastic-agent-managed/kustomization.yaml):
+
+- %ES_HOST%: The Elasticsearch host to communicate with
+- %API_KEY: The API Key with access privilleges to connect to Elasticsearch. See [create-api-key-standalone-agent](https://www.elastic.co/guide/en/fleet/current/grant-access-to-elasticsearch.html#create-api-key-standalone-agent). *This should be encoded as base64 value because it will be stored as Kubernetes secret*
+- %CA_TRUSTED%: The ssl.ca_trusted_fingerprint in order the elastic agent to be able to trust specific certificate
+- %DATASET_ID%: A string that will be added as a new field and will denote a specific installation. *By default, this will be added to state_pod dataset.*
```

Comment:
I don't understand what this does. Is this dataset in the context of the data stream naming scheme? Why is there only one?

Reply:
In the standalone agent installations we can not verify if agents is installed and when is ready. So when a user through kibana will start a new installation (so it will have a specific session), we will include this session in the field dataset_id. And then kibana will have a loop to check whether this field has been created. 

FYI see https://github.com/elastic/elastic-agent/issues/4657 - the [Stretched goal]

Reply:
I name that doesn't repurpose the dataset terminology would make this clearer.

Also this is effectively an agent ID that isn't set by Fleet? We could just add the ability to set that so that it would populate as the `agent.id` field in documents instead of us generating a random one at startup.

Reply:
> name that doesn't repurpose the dataset terminology would make this clearer.

@flash1293 , @mykolaharmash any other ideas regarding variable name? 

@cmacknz actually yes this is an agentId for standalone scenarios. 

> We could just add the ability to set that so that it would populate as the agent.id

When this is ready, I agree we can remove this workaround. This is I guess a quick solution now to make this working. Do you have some implementation in mind?

Reply:
It's not really an agent id, because it will be the same on every node of the cluster - by my understanding, the individual nodes will get assigned separate agent ids.

I agree that "dataset" is a bit confusing too, what about "onboarding id" - that's what it is after all

Reply:
Anything that isn't already a term used elsewhere in the product is fine with me.

Reply:
Thank you both, ONBOARDING_ID added in 
[89cc0d9](https://github.com/elastic/elastic-agent/pull/4754/commits/89cc0d9062d59634111d18f6a83353f75783ef56)